server/mailet/mailets/src/main/java/org/apache/james/transport/mailets/RecipientRewriteTableProcessor.java

```diff
@@ -223,37 +223,27 @@ static ForwardDecision removeRecipient(MailAddress recipient) {
                 .collect(ImmutableList.toImmutableList()));
         }
 
-        static ForwardDecision sendACopy(MailetContext context, MailAddress originalRecipient, List<MailAddress> forwards) {
+        static ForwardDecision sendACopy(MailetContext context,
+                                         MailAddress originalRecipient,
+                                         List<MailAddress> forwards,
+                                         LoopPrevention.RecordedRecipients recordedRecipients,
+                                         Set<MailAddress> newRecipients) {
```

Comment:
Way too much arguments. With 4 different disguised list of mail address. 
That's not reasonnable.

Reply:
removed one argument